lib/active_merchant/billing/gateways/plexo.rb

```diff
@@ -110,13 +99,28 @@ def encoded_credentials
         Base64.encode64("#{@credentials[:client_id]}:#{@credentials[:api_key]}").delete("\n")
       end
 
+      def build_auth_purchase_request(money, post, payment, options)
+        post[:ReferenceId] = options[:reference_id] || generate_unique_id
+        post[:MerchantId] = options[:merchant_id] || @credentials[:merchant_id]
+        post[:Installments] = options[:installments] if options[:installments]
+        post[:StatementDescriptor] = options[:statement_descriptor] if options[:statement_descriptor]
+        post[:CustomerId] = options[:customer_id] if options[:customer_id]
+
+        add_payment_method(post, payment, options)
+        add_items(post, options[:items])
+        add_metadata(post, options[:metadata])
+        add_amount(money, post, options)
+        add_browser_details(post, options)
+      end
+
       def header(parameters = {})
-        {
+        headers = {
           'Content-Type' => 'application/json',
-          'Authorization' => "Basic #{encoded_credentials}",
-          'x-mock-tokenization' => parameters.dig(:header, :mock_tokenization) || 'true',
-          'x-mock-switcher' => parameters.dig(:header, :mock_switcher) || 'true'
+          'Authorization' => "Basic #{encoded_credentials}"
         }
+        headers['x-mock-tokenization'] = parameters.dig(:header, :mock_tokenization) if parameters.dig(:header, :mock_tokenization)
+        headers['x-mock-switcher'] = parameters.dig(:header, :mock_switcher) if parameters.dig(:header, :mock_switcher)
```

Comment:
Thanks for brining my attention to that line to remove the headers as I was following the description on the ticket, I have updated the code so it doesn't contain them.

For the verify and declined transactions issue, I was of the opinion that this change is not related to this ticket so we could be ready to merge them as it would probably be fixed from gateway or else we can do them in separate ticket but I am also okay to wait for plexo response and get them pass in this PR as well 👍